### DISCUSSION TITLE
Weird i2c (default wire library) behavior

### DISCUSSION THREAD

**Discussion by dattasaurabh82** [SpenceKonde/megaTinyCore#465] [Library discussion]
So I was trying to write a small scanner because I wanted to use an RTC [RV-8803-C7](https://www.microcrystal.com/fileadmin/Media/Products/RTC/App.Manual/RV-8803-C7_App-Manual.pdf)

pin map:
**RV-8803-C7** -------- **ATTINY1607**
VIN<----------------- SUPPLY VOLTAGE (3V)
GND<---------------- GND
SCL<---------------- SCL
SDA<---------------- SDA

I came across [tiny-mega-i2c](https://github.com/technoblogy/tiny-mega-i2c) as an alt to wire library for it's low footprint. 

For a simple scanner code, using  [tiny-mega-i2c](https://github.com/technoblogy/tiny-mega-i2c), I get the proper address on scan:
```
#include <TinyMegaI2CMaster.h>

void setup() {
  TinyMegaI2C.init();

  Serial.begin(115200);
  Serial.println(F("\nI2C Scanner"));
  delay(1000);
}

void loop() {
  byte address;
  int nDevices;

  Serial.println(F("Scanning..."));
  nDevices = 0;

  for (address = 1; address < 127; address++ ) {
    if  (TinyMegaI2C.start(address, 0)) {
      nDevices++;
      Serial.print(F("I2C device found at address 0x"));
      Serial.print(address, HEX);
      Serial.println(F("  !"));
    }
  }
  if (nDevices == 0)
    Serial.println(F("No I2C devices found\n"));
  else
    Serial.println(F("done\n"));
  delay(2000);
}
```

If I now try to write the same scanner with std. wire library, nothing happens 🤨
```
#include <Wire.h>

void setup() {
  Wire.begin();

  Serial.begin(115200);
  Serial.println(F("\nI2C Scanner"));
  delay(1000);
}

void loop() {
  byte error, address;  //variable for error and I2C address
  int nDevices;

  Serial.println(F("Scanning..."));
  nDevices = 0;for (address = 1; address < 127; address++ )
    {
      // The i2c_scanner uses the return value of
      // the Write.endTransmisstion to see if
      // a device did acknowledge to the address.
      Wire.beginTransmission(address);
      error = Wire.endTransmission();
  
      if (error == 0)
      {
        Serial.print(F("I2C device found at address 0x"));
        if (address < 16)
          Serial.print(F("0"));
        Serial.print(address, HEX);
        Serial.println(F("  !"));
  
        nDevices++;
      }
      else if (error == 4)
      {
        Serial.print(F("Unknown error at address 0x"));
        if (address < 16)
          Serial.print(F("0"));
        Serial.println(address, HEX);
      }
    }
    if (nDevices == 0)
      Serial.println(F("No I2C devices found\n"));
    else
      Serial.println(F("done\n"));
  
    delay(2000);  
}
```

Is this because of the **non-existence** of `external pull-ups` on my connections, or something else? 
Now I'm wondering how some [other slave libraries that uses std wire library](https://github.com/SpenceKonde/megaTinyCore/blob/master/megaavr/extras/LibraryCompatibility.md) are working with the new ATTINYs! ☹️

**Comment by SpenceKonde** [comment]
That is the expected result if you don't have external pullups. This described in the Wire/I2C section of the readme....

**Reply by dattasaurabh82** [reply]
@SpenceKonde After the pull-ups, the i2cScanner using std wire lib works without any hiccups.  

![WechatIMG566](https://user-images.githubusercontent.com/4619862/122935564-0449af80-d3a3-11eb-8a36-5a67c5192b8f.jpeg) ![WechatIMG567](https://user-images.githubusercontent.com/4619862/122935591-0c095400-d3a3-11eb-9d73-804af24600c4.jpeg)

But using std wire lib still uses double the memory as compared to [tiny-mega-i2c](https://github.com/technoblogy/tiny-mega-i2c)
You have contributed to that library as well it seems 🤗
